tests/ui/result_unit_error.rs

```diff
@@ -0,0 +1,34 @@
+#[warn(clippy::result_unit_err)]
+#[allow(unused)]
+
+pub fn returns_unit_error() -> Result<u32, ()> {
+    Err(())
+}
+
+fn private_unit_errors() -> Result<String, ()> {
+    Err(())
+}
+
+pub trait HasUnitError {
+    fn get_that_error(&self) -> Result<bool, ()>;
+}
```

Comment:
```suggestion

    fn get_this_one_too(&self) -> Result<bool, ()> {
        Err(())
    }
}
```
We could check that provided trait definitions are linted too.